apps/src/code-studio/components/ExportDialog.jsx

```diff
@@ -12,6 +12,8 @@ import exportExpoIconPng from '../../templates/export/expo/icon.png';
 import SendToPhone from './SendToPhone';
 import project from '../initApp/project';
 
+const APK_BUILD_STATUS_CHECK_PERIOD = 60000;
```

Comment:
Given how much this PR is changing ExportDialog, this seems like a good time to refactor ExportDialog.jsx
Couple of thoughts (I didn't look too closely, so these are only starting points):
- ~1000 lines in a react component, while not necessarily a bad thing, does give me pause as it is usually indicative of code smell.
- cancelIfGeneratingAndResetState and cancelIfPreexistingApkBuild do very similar things. 
- publishExpoExport and publishAndGenerateApk seem like they should either be merged or should be split into different react components
- renderIntroPage, renderPlatformPage, renderIconPage, renderPublishPage, renderGeneratingPage seem like they could all be separate components 
- the render function is difficult to digest in one sitting - a lot of it is conditionally rendered

Reply:
We can create some other components for each page, that was discussed in the previous PR where I created `ExportDialog`

`cancelIfGeneratingAndResetState()` deals with activities that the current dialog controls (and has knowledge of through its state)

`cancelIfPreexistingApkBuild()` deals with activities that happen prior to the lifetime of this page loading.

`publishExpoExport()` is still used by `AdvancedShareOptions`. Until we have removed all Expo functionality from there, it is difficult to refactor. I'm trying to share the code to avoid duplication.

Reply:
Would it make more sense to split out the things used by AdvancedShareOptions?

Reply:
Talked offline: Chris is going to do a follow-up PR to refactor ExportDialog